Remove some unnecessary fields from minidumps

hlinnaka opened the pull request:
CDXLDatum's IsByValue flag was not used for anything. Remove it.

Don't serialize `IsNull="false"` attribute for constants. It's redundant with the fact that non-NULLs have the 'Value' attribute. Only dump `IsNull`, when the constant is a NULL. Makes minidumps somewhat smaller.

Together these two changes shrink the total size of `data` directory from 84  MB to 75 MB.

asubramanya commented:
@hlinnaka Thanks for your PR. 

I had a few issues compiling the original PR. For e.g
```
../libnaucrates/src/md/CMDTypeBoolGPDB.cpp:255:25: error: no member named 'IsPassedByValue' in 'gpdxl::CDXLDatumBool'
        GPOS_ASSERT(dxl_datum->IsPassedByValue());
                    ~~~~~~~~~  ^
../libgpos/include/gpos/assert.h:35:42: note: expanded from macro 'GPOS_ASSERT'
#define GPOS_ASSERT(x)                  GPOS_RTL_ASSERT(x)
                                                        ^
../libgpos/include/gpos/assert.h:28:31: note: expanded from macro 'GPOS_RTL_ASSERT'
#define GPOS_RTL_ASSERT(x)              ((x) ? ((void)0) : \
                                          ^
../libnaucrates/src/md/CMDTypeBoolGPDB.cpp:277:30: error: no member named 'IsPassedByValue' in 'gpdxl::CDXLDatumBool'
        GPOS_ASSERT(dxl_datum_bool->IsPassedByValue());
```

I fixed those by getting rid of the asserts. But after that some tests started failing. I fixed those as well. You probably built a non debug build of ORCA and thats why you did not see these errors.
I also rebased the branch off of ORCA master and got the tests to pass. All these changes are in my private branch.

I tried to compile this branch with GPDB and I see some failures.
```
CTranslatorDXLToScalar.cpp:1439:40: error: no member named 'IsPassedByValue' in 'gpdxl::CDXLDatumOid'
        constant->constbyval = oid_datum_dxl->IsPassedByValue();
                               ~~~~~~~~~~~~~  ^
CTranslatorDXLToScalar.cpp:1475:41: error: no member named 'IsPassedByValue' in 'gpdxl::CDXLDatumInt2'
        constant->constbyval = datum_int2_dxl->IsPassedByValue();
                               ~~~~~~~~~~~~~~  ^
CTranslatorDXLToScalar.cpp:1511:41: error: no member named 'IsPassedByValue' in 'gpdxl::CDXLDatumInt4'
        constant->constbyval = datum_int4_dxl->IsPassedByValue();
```

Seems like we need to change the translators on the GPDB side too. I am not sure if you have a PR open for that on the GPDB side, I did not see one there. Can you take a look at the GPDB side changes?

hlinnaka commented:
On 15/05/2019 03:59, Abhijit Subramanya wrote:
> @hlinnaka <https://github.com/hlinnaka> Thanks for your PR.
> 
> I had a few issues compiling the original PR. For e.g
> 
> ... > I also rebased the branch off of ORCA master and got the tests to pass.
> All these changes are in my private branch.

Thanks, I incorporated those changes in the PR.

> I tried to compile this branch with GPDB and I see some failures.
> 
> |CTranslatorDXLToScalar.cpp:1439:40: error: no member named 
> 'IsPassedByValue' in 'gpdxl::CDXLDatumOid' constant->constbyval = 
> oid_datum_dxl->IsPassedByValue(); ~~~~~~~~~~~~~ ^ 
> CTranslatorDXLToScalar.cpp:1475:41: error: no member named 
> 'IsPassedByValue' in 'gpdxl::CDXLDatumInt2' constant->constbyval = 
> datum_int2_dxl->IsPassedByValue(); ~~~~~~~~~~~~~~ ^ 
> CTranslatorDXLToScalar.cpp:1511:41: error: no member named 
> 'IsPassedByValue' in 'gpdxl::CDXLDatumInt4' constant->constbyval = 
> datum_int4_dxl->IsPassedByValue(); |
> 
> Seems like we need to change the translators on the GPDB side too. I am 
> not sure if you have a PR open for that on the GPDB side, I did not see 
> one there. Can you take a look at the GPDB side changes?

Ah yeah, I hadn't done that yet. Opened 
https://github.com/greenplum-db/gpdb/pull/7687 for that now.

- Heikki

bhuvnesh2703 commented:
LGTM.
@asubramanya should this go in 5x? or we should fork orca now for 6 and master..?
